src/Extensions/AzureBlobStorage/Program.cs

```diff
@@ -0,0 +1,126 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Diagnostics.Tools.Monitor.Egress;
+using Microsoft.Diagnostics.Tools.Monitor.Egress.AzureBlob;
+using System.CommandLine;
+using System.CommandLine.Binding;
+using System.Text.Json;
+
+namespace Microsoft.Diagnostics.Monitoring.AzureStorage
+{
+    internal class Program
+    {
+        private static Stream StdInStream = null;
+        private static CancellationTokenSource CancelSource = new CancellationTokenSource();
+        static async Task<int> Main(string[] args)
+        {
+            // Expected command line format is: AzureBlobStorage.exe Egress --Provider-Name MyProviderEndpointName
+            RootCommand rootCommand = new RootCommand("Egresses an artifact to Azure Storage.");
+
+            Command egressCmd = new Command("Egress", "The class of extension being invoked; Egress is for egressing an artifact.");
+            egressCmd.Add(new Option<string>("--Provider-Name", "The provider name given in the configuration to dotnet-monitor."));
+            egressCmd.SetHandler<string>(Program.Egress, egressCmd.Children.OfType<IValueDescriptor>().ToArray());
+            rootCommand.Add(egressCmd);
+
+            return await rootCommand.InvokeAsync(args);
+        }
+
+        private static async Task<int> Egress(string ProviderName)
+        {
+            EgressArtifactResult result = new();
+            try
+            {
+                string jsonConfig = Console.ReadLine();
+                ExtensionEgressPayload configPayload = JsonSerializer.Deserialize<ExtensionEgressPayload>(jsonConfig);
+                AzureBlobEgressProviderOptions options = BuildOptions(configPayload);
+
+                AzureBlobEgressProvider provider = new AzureBlobEgressProvider(logger: null); // TODO: Replace logger with real instance of console logger (console events on standard out will get forwarded to dotnet-monitor and written to it's console).
+
+                Console.CancelKeyPress += Console_CancelKeyPress;
+
+                result.ArtifactPath = await provider.EgressAsync(EgressProviderTypes.AzureBlobStorage, configPayload.ProfileName, options, GetStream, configPayload.Settings, CancelSource.Token);
+                result.Succeeded = true;
+            }
+            catch (Exception ex)
+            {
+                result.Succeeded = false;
+                result.FailureMessage = ex.Message;
+            }
+
+            string jsonBlob = JsonSerializer.Serialize<EgressArtifactResult>(result);
+            Console.Write(jsonBlob);
```

Comment:
Do we need to flush?

Reply:
No, this will always get fully written to `StdOut`